component/src/test/java/org/wso2/extension/siddhi/io/websocket/sink/util/ResultContainer.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.extension.siddhi.io.websocket.sink.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Class to retain results received by JMS client so that tests can poll the result and assert against.
+ */
+public class ResultContainer {
+    private static Log log = LogFactory.getLog(ResultContainer.class);
+    private int eventCount;
+    private List<String> results;
+    private CountDownLatch latch;
+    private int timeout = 90;
+
+
+    public ResultContainer(int expectedEventCount) {
+        eventCount = 0;
+        results = new ArrayList<>(expectedEventCount);
+        latch = new CountDownLatch(expectedEventCount);
+    }
+
+    public void eventReceived(String message) {
+        eventCount++;
+        results.add(message);
+        latch.countDown();
+    }
+
+    public void eventReceived(byte[] message) throws UnsupportedEncodingException {
+        eventCount++;
+        String text = new String(message, "utf-8");
```

Comment:
This `String` constructor throws a `UnsupportedEncodingException` exception. Instead use `String(byte bytes[], Charset charset)` constructor. Sample [here](https://github.com/this/carbon-ui-server/blob/ca3bf71389b28a04c5ea3a9038dea423d9711fc1/components/org.wso2.carbon.uis/src/main/java/org/wso2/carbon/uis/internal/io/reference/ArtifactFileReference.java#L61)

Reply:
fixed in 9fea7d3